wp-admin/js/widgets/media-widgets.js

```diff
@@ -429,21 +428,26 @@ wp.mediaWidgets = ( function( $ ) {
 		 * @returns {void}
 		 */
 		selectMedia: function selectMedia() {
-			var control = this, selection, mediaFrame, defaultSync;
+			var control = this, selection, mediaFrame, defaultSync, mediaFrameProps;
 
 			if ( control.isSelected() && 0 !== control.model.get( 'attachment_id' ) ) {
 				selection = new wp.media.model.Selection( [ control.selectedAttachment ] );
 			} else {
 				selection = null;
 			}
 
+			mediaFrameProps = control.mapModelToMediaFrameProps( control.model.toJSON() );
+			control.displaySettings.set( 'size', mediaFrameProps.size );
```

Comment:
Isn't `size` image specific?  Could we just check the presence of `mediaFrameProps.size` prior to this line?